File: 2444-count-subarrays-with-fixed-bounds/2444-count-subarrays-with-fixed-bounds.py
class Solution:
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        ans = 0
        minPosition = -1
        maxPosition = -1
        leftBound = -1

        for i in range(len(nums)):
            if nums[i] < minK or nums[i] > maxK:
                leftBound = i

            if nums[i] == minK:
                minPosition = i
            if nums[i] == maxK:
                maxPosition = i

            count = min(maxPosition, minPosition) - leftBound

            ans += 0 if count <= 0 else count

        return ans

File: LeetCode/2444-count-subarrays-with-fixed-bounds/2444-count-subarrays-with-fixed-bounds.java
class Solution {
    public long countSubarrays(int[] nums, int minK, int maxK) {
        long ans = 0;
        int minPosition = -1;
        int maxPosition = -1;
        int leftBound = -1;

        for (int i = 0; i < nums.length; i++) {

            if (nums[i] < minK || nums[i] > maxK)
                leftBound = i;

            if (nums[i] == minK)
                minPosition = i;
            if (nums[i] == maxK)
                maxPosition = i;

            int count = Math.min(maxPosition, minPosition) - leftBound;

            ans += (count <= 0) ? 0 : count;

        }

        return ans;
    }
}